portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,30 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+function msg(){
+  var text = prompt("Leave Justine a message: ");
+  var name = prompt("Your name: ");
+  if (name != null && name.length>0){
+    alert("Thanks for your message "+name+ "! Justine will get back to you shortly.");
+  }
+}
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+function bunnymsg(){
+  var bunny=document.getElementById('bunny');
+  document.getElementById("bun_msg").innerHTML="*bunny noises*";
```

Comment:
If you don't need to write HTML, it's better to use `textContent` or `innerText`.

https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent

If you do need to build out HTML within a node, you should still *not* use innerHTML, because that's a great way to cause a cross site scripting vulnerability; that is, imagine you were displaying user-generated content like comments. suppose you created a comment like this:

`commentParentElement.innerHTML = "<div>" + comment + "</div>";`

Think about what happens if someone writes on your site a comment that contains a `<script>` tag; that user can then write code that executes whenever *other* users visit that page on your site, and that code behaves as if it is part of your site, so it can take actions that impersonate the other user.

Normally you would use a framework that handles building out pages from templates, but in plain JavaScript, what you can do is build out subtrees of tags using `document.createElement` and `element.appendChild`.

Reply:
Thanks, that makes sense. Most tutorials don't go into why .innerHTML is problematic.